resource-managers/yarn/pom.xml

```diff
@@ -30,8 +30,18 @@
   <properties>
     <sbt.project.name>yarn</sbt.project.name>
     <jersey-1.version>1.19</jersey-1.version>
+    <spark.yarn.isHadoopProvided>false</spark.yarn.isHadoopProvided>
   </properties>
 
+  <profiles>
+    <profile>
+      <id>hadoop-provided</id>
+      <properties>
+        <spark.yarn.isHadoopProvided>true</spark.yarn.isHadoopProvided>
+      </properties>
+    </profile>
+  </profiles>
+
```

Comment:
Do we need to mention this profile option in docs?

Reply:
Seems we already have https://github.com/apache/spark/blob/b11e42663be680a0357f2bf7bd8b16afe313eb5e/docs/building-spark.md#packaging-without-hadoop-dependencies-for-yarn

Reply:
This profile is already in parent POM, so when we build it with `hadoop-provided` from the root project, this will only add the property to yarn scope as we don't need this property globally for the entire Spark project.

Reply:
Yea, thanks. I see. I just wonder if we want to mention this profile in the docs you added (docs/running-on-yarn.md), or adding a link to the doc link I posted above, so users can know how to build a no-hadoop distribution, when reading this.

Reply:
The next line of added doc `To build Spark yourself, refer to [Building Spark](building-spark.html).` has the link to the doc you mentioned. I think they should be able to figure it out when they read building spark session.